# Cry1Ac-PACE-paper
Code for analysis of NGS data from Cry1Ac - Soybean Looper Cadherin PACE experiment. The following is a relevant section from the methods of our paper, which can be found at [http://www.nature.com/nature/journal/vaop/ncurrent/full/nature17938.html](http://www.nature.com/nature/journal/vaop/ncurrent/full/nature17938.html)

## High-throughput sequencing and oligotype analysis
Raw reads have been deposited in the NCBI Sequence Read Archive under accession number PRJNA293870, and all custom scripts used in analysis are available at [http://github.com/MonsantoCo/BadranEtAl2015](http://github.com/MonsantoCo/BadranEtAl2015). Illumina reads obtained from each time point were mapped to the SP055-rpoZ-cMyc-Cry1Ac1-d123 reference sequence using [bowtie](http://bowtie-bio.sourceforge.net/bowtie2/index.shtml) version 2.1.0, and the resulting SAM files were combined into a single BAM file using [samtools](http://samtools.sourceforge.net/) version 0.1.19. This BAM file was used as input to [freebayes](http://github.com/ekg/freebayes) version 0.9.21-12-g92eb53a36 to call single nucleotide polymorphisms, using the command `freebayes–use-best-n-alleles 1–pooled-continuous–use-reference-allele–theta 500000000–min-alternate-fraction 0.01–ploidy 1–region SP055-rpoZ-cMyc-Cry1Ac1-d123:2833-4971`. The analysis is encapsulated in the custom script [ill.callsnps.sh](http://github.com/MonsantoCo/BadranEtAl2015/blob/master/sh/ill.callsnps.sh). PacBio polymerase reads were demultiplexed with `RS_Resequencing_Barcode.1` workflow provided by PacBio. Polymerase reads with quality score lower than 0.80 (defined by the PacBio scoring algorithm) or shorter than 50 base pairs were filtered. High-quality reads were processed into subreads after sequencing primers and adaptors were removed. Circular consensus reads (or reads-of-inserts) were obtained by calling consensus of subreads generated from the same polymerase reads. These circular consensus reads were mapped to the SP055-rpoZ-cMyc-Cry1Ac1-d123 reference sequence using [BLASR](http://github.com/PacificBiosciences/blasr) version 1.3.1.142244, and the alignment was exported as an aligned FASTA sequence using the custom script [SAMtoAFA.py](http://github.com/MonsantoCo/BadranEtAl2015/blob/master/SAMtoAFA.py). The aligned FASTA was used as input to the [oligotyping](http://github.com/meren/oligotyping) platform, manually specifying entropy components as the positions at which the Illumina data defined informative single nucleotide polymorphisms. Only oligotypes that occur at >1% in at least one sample were retained. This methodology resulted in informative changes at 25 of the 27 specified components. Oligotypes with gaps at the specified components, probably because of indels in the PacBio sequencing or alignment, were reassigned to other oligotypes with nucleotides in those positions only when it could be done unambiguously, and discarded otherwise, resulting in a total fraction abundance <1 in Fig. 5d. The resulting oligotype-percent abundance matrix was read into R and analysed using the custom script [PedigreeAndMullerPlot.R](http://github.com/MonsantoCo/BadranEtAl2015/blob/master/R/PedigreeAndMullerPlot.R). The pedigree was refined manually, assuming that single-mutant derivatives of previous oligotypes were due to de novo mutation, while double, triple, or greater mutations that can be explained by recombination of previously observed oligotypes were due to recombination, since these last types of mutation were highly unlikely to arise by multiple point mutation after the start of the PACE experiment.
